### ISSUE TITLE
Carbon::createFromFormat using a date without time does not respect Carbon::setTestNow

### ISSUE THREAD

**Issue by Naouak** [briannesbitt/Carbon#1280]
When using createFromFormat without a time, instead of using the time fixed by setTestNow, it use current time of the server which is not the expected behavior.

To reproduce:
```php
<?php
require('vendor/autoload.php');
Carbon\Carbon::setTestNow(Carbon\Carbon::createFromFormat("Y-m-d H:i:s", '2018-01-01 00:00:00'));
$date = Carbon\Carbon::createFromFormat("Y-m-d", "2018-05-06");
$date->hour == 0 || die("Invalid time");
?>
```

**Comment by Glavic** [comment]
> which is not the expected behavior.

PHP manual states that this **is** the default behaviour...
> If format does not contain the character ! then portions of the generated time which are not specified in format will be set to the current system time.

Read more here: http://php.net/manual/en/datetime.createfromformat.php

On the link look at the format character `!`:
> Resets all fields (year, month, day, hour, minute, second, fraction and timezone information) to the Unix Epoch

Based on that:

```
$date = Carbon\Carbon::createFromFormat("!Y-m-d", "2018-05-06");
```

**Comment by kylekatarnls** [comment]
Will be fixed in 1.28 thanks to @imbrish #1281
